api/app/services/repository_sync_service.rb

```diff
@@ -7,12 +7,12 @@ def index
     gitland_repository = Gitland::Repository.new(@repository)
     gitland_repository.pull
 
-    @repository.transaction do
-      extract_full_commit_history(gitland_repository)
-      extract_commit_history_for_changes_ledger(gitland_repository)
+    latest_known_commit = @repository.commits.last
```

Comment:
This is the fix. Instead of resolving the `latest known commit` in both steps, we resolve it once at the beginning of the sync.